### PR TITLE
docgen: Add function to handle list of types for theme.json schema

### PR DESCRIPTION
**Pull request by mikachan**
## What?

This PR enables docgen to list multiple types for the theme.json style properties listed in theme-json-living.md.

## Why?

Currently, the generated theme.json docs will not list multiple types and instead will print `undefined`. See the [current generated entry](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/theme-json-reference/theme-json-living.md#border-1) for the `radius` style property.

## How?

Adds a new function called `formatType`, which takes the style property object and checks if the `type` key exists. If it does, it will default to that (the same as the current behavior). If it doesn't exist, it checks for the `anyOf` or `oneOf` properties and makes a comma-separated list of those types.

## Testing Instructions

Run `node ./bin/api-docs/gen-theme-reference.js`, and the generated type for the `radius` property should be "string, object" instead of "undefined".

### CODE REVIEW

File: bin/api-docs/gen-theme-reference.js [hunk 1]
```diff
@@ -160,6 +161,23 @@ ${ markupFn( data ) }
...
+ * @param {Object} prop
+ * @return {string} type
+ */
+const formatType = ( prop ) => {
+	let type = prop.type || '';
+
+	if ( prop.hasOwnProperty( 'anyOf' ) || prop.hasOwnProperty( 'oneOf' ) ) {
+		const types = prop.anyOf || prop.oneOf;
+		type = types.map( ( item ) => item.type ).join( ', ' );
+	}
```
Comment (vcanales) [inline bin/api-docs/gen-theme-reference.js]: `formatType` is currently adding a trailing comma when extracting types from `$ref` properties as they don't include a `type` property.

```js
[ { type: 'string' }, { '$ref': '#/definitions/refComplete' } ]
/*
  ### filter

CSS and SVG filter styles.

| Property | Type    | Props |
| -------- | ------- | ----- |
| duotone  | string, |       |
*/
```

Filtering out entries that don't include `type` fixes this.

```suggestion
	if ( prop.hasOwnProperty( 'anyOf' ) || prop.hasOwnProperty( 'oneOf' ) ) {
		const types = prop.anyOf || prop.oneOf;
		type = types
			.filter( ( item ) => !! item.type )
			.map( ( item ) => item.type )
			.join( ', ' );
	}
```

We might want to add a way to describe the `$ref` property, but that's probably better left for later.

**Review by vcanales — CHANGES_REQUESTED** [review]

File: bin/api-docs/gen-theme-reference.js (hunk 1 above)
Comment (mikachan) [reply bin/api-docs/gen-theme-reference.js]: Ah yeah good spot, I guess I was thinking of tackling this when the $ref type had been added. 

$ref will always be an object, so maybe if $ref exists, we could just list 'object'? I've pushed a change that does this to see what you think 🙏

Comment (vcanales) [reply bin/api-docs/gen-theme-reference.js]: If there's no specific type for `$ref`, it should be fine to use `object`

**Review by vcanales — APPROVED** [review]
LGTM

**Pull request merged**